mobile/native/src/db/custom_types.rs

```diff
@@ -161,20 +150,27 @@ impl ToSql<Text, Sqlite> for FailureReason {
 impl FromSql<Text, Sqlite> for FailureReason {
     fn from_sql(bytes: backend::RawValue<Sqlite>) -> deserialize::Result<Self> {
         let string = <String as FromSql<Text, Sqlite>>::from_sql(bytes)?;
-
-        return match string.as_str() {
-            "TradeRequest" => Ok(FailureReason::TradeRequest),
-            "TradeResponse" => Ok(FailureReason::TradeResponse),
-            "ProposeDlcChannel" => Ok(FailureReason::CollabRevert),
-            "FailedToSetToFilling" => Ok(FailureReason::FailedToSetToFilling),
-            "OrderNotAcceptable" => Ok(FailureReason::OrderNotAcceptable),
-            "TimedOut" => Ok(FailureReason::TimedOut),
-            "SubchannelOfferOutdated" => Ok(FailureReason::SubchannelOfferOutdated),
-            "SubchannelOfferDateUndetermined" => Ok(FailureReason::SubchannelOfferDateUndetermined),
-            "SubchannelOfferUnacceptable" => Ok(FailureReason::SubchannelOfferUnacceptable),
-            "OrderRejected" => Ok(FailureReason::OrderRejected),
-            _ => Err("Unrecognized enum variant".into()),
-        };
+        match serde_json::from_str(string.as_str()) {
+            Ok(reason) => Ok(reason),
+            Err(_) => {
+                // backwards compatibility
```

Comment:
🔧 Do we need backwards compatibility here? Maybe just say "unknown" for all of them.